config/settings.d/puppetca.yml.example

```diff
@@ -2,7 +2,9 @@
 # Can be true, false, or http/https to enable just one of the protocols
 :enabled: false
 
+# valid providers:
+#   - puppetca_hostverify (verify CSRs based on hostnames)
```

Comment:
I don't like `basic autosigning` because it says nothing about how it works, `autosignfile` is better, but because the token based approach will also use a file it is not clear which is which. The thing that will distinguish the providers is the attribute which is compared during autosigning which is `certname` vs. `token`.
So imho the best name would be `puppetca_hostnamebasedautosigning` (or `certnamebased`) but this is a pain to read.

Reply:
How about `HostnameWhitelisting`?

Reply:
I like that, but wouldn't we want to use snake_case?

Reply:
Yep, 🐍.